akka-stream/src/main/scala/akka/stream/impl/io/TcpStages.scala

```diff
@@ -272,12 +274,6 @@ private[stream] object TcpConnectionStage {
 
       override def onUpstreamFailure(ex: Throwable): Unit = {
         if (connection != null) {
-          if (interpreter.log.isDebugEnabled) {
```

Comment:
this was explicitly added due to the request by @jroper back then, and it makes sense. You really want to know why a connection was terminated on failures, not just have it silently dropped, at least in DEBUG level.